src/humanloop/otel/exporter/proto.py

```diff
@@ -0,0 +1,76 @@
+                        ),
+                        spans=[
+                            ProtoBufferSpan(
+                                trace_id=span_to_export.context.trace_id.to_bytes(length=16, byteorder="big"),
+                                span_id=span_to_export.context.span_id.to_bytes(length=8, byteorder="big"),
+                                name=span_to_export.name,
+                                kind={
+                                    0: ProtoBufferSpan.SpanKind.SPAN_KIND_INTERNAL,
+                                    1: ProtoBufferSpan.SpanKind.SPAN_KIND_SERVER,
+                                    2: ProtoBufferSpan.SpanKind.SPAN_KIND_CLIENT,
+                                    3: ProtoBufferSpan.SpanKind.SPAN_KIND_PRODUCER,
+                                    4: ProtoBufferSpan.SpanKind.SPAN_KIND_CONSUMER,
+                                }[span_to_export.kind.value],
+                                start_time_unix_nano=span_to_export.start_time,  # type: ignore [attr-defined, arg-type]
+                                end_time_unix_nano=span_to_export.end_time,  # type: ignore [attr-defined, arg-type]
+                                attributes=[
+                                    KeyValue(
+                                        key=key,
+                                        value=AnyValue(string_value=str(value)),
+                                    )
+                                    for key, value in span_to_export.attributes.items()  # type: ignore [union-attr]
+                                ],
+                                dropped_attributes_count=span_to_export.dropped_attributes,
+                                dropped_events_count=span_to_export.dropped_events,
+                                dropped_links_count=span_to_export.dropped_links,
+                                links=[
+                                    ProtoBufferSpan.Link(
+                                        trace_id=link.context.trace_id.to_bytes(length=16, byteorder="big"),
+                                        span_id=link.context.span_id.to_bytes(length=8, byteorder="big"),
+                                        attributes=[
+                                            KeyValue(
+                                                key=key,
+                                                value=AnyValue(string_value=str(value)),
+                                            )
+                                            for key, value in link.attributes.items()  # type: ignore [union-attr]
+                                        ],
+                                    )
+                                    for link in span_to_export.links
+                                ],
+                                events=[],
+                            )
+                        ],
+                    )
+                ]
+            )
+        ]
+    )
+
+    return MessageToJson(payload)
```

Comment:
hmm not 100% sure this will work as expected -- OTLP JSON encoding differs a bit from the proto3 standard mapping. Possible we should just be working in proto anyways? Might make things simpler since it's more of the default rather than JSON.

https://opentelemetry.io/docs/specs/otlp/#json-protobuf-encoding

Reply:
I guess it's fine if our IDs aren't "correct" as long as they're internally consistent?